content/blog/2019/10/2019-10-04-vscode-caseStudy.adoc

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "JCasC JSONSchema Case-Study Using VSCode"
+opengraph:
+  image: /images/logos/JCasC/JCasC.png
+tags:
+- community-bridge
+- JCasC
+- VSCode
+- community
+author: sladyn98
+---
+
+=== JSON Schema and VSCode
+
+==== Configuration-as-code plugin
+The Configuration as Code plugin has been designed as an opinionated way to configure Jenkins based on human-readable declarative configuration files. Writing such a file should be feasible without being a Jenkins expert, just translating into code a configuration process one is used to executing in the web UI.The plugin uses a schema to verify the files being applied to the Jenkins instance.
+
+With the new JSON Schema being enabled developers can now test their yaml file against it. The new schema checks for the configurators, their types and to a certain extent the nesting of the configurators. VSCode allows us to test out the schema right out of the box with some modifications.
+
+=== Steps to Enable the Schema Validation
+
+a) The first step includes installing the YAML plugin from redhat. Open up the Extension list in VSCode editor using `Ctrl + Shift + X`.
+
+b) Select and install the YAML plugin by Red Hat. Reload the editor if necessary.
+
+c) The next step includes inserting the JSON Schema inside the VSCode workspace. We would be including the schema in the `work` directory.
```

Comment:
why in the work directory? most users won't be running jenkins via `hpi:run`

Reply:
Where do you want it, in the base directory ?

Reply:
current directory?

Reply:
Done :)